tf_agents/cogment_verse_tf_agents/reinforce/reinforce.py

```diff
@@ -30,48 +31,42 @@ class ReinforceAgent:
     # pylint: disable=too-many-instance-attributes
     def __init__(
         self,
-        id,
-        obs_dim,
-        act_dim,
-        epsilon=0.01,
-        gamma=0.95,
-        lr=3e-4,
-        max_replay_buffer_size=50000,
-        seed=42,
-        epsilon_schedule=None,
-        lr_schedule=None,
+        **params
     ):
-        self.id = id
-        self._params = {}
-        self._params["obs_dim"] = obs_dim
-        self._params["act_dim"] = act_dim
-        self._params["seed"] = seed
-        self._params["max_replay_buffer_size"] = max_replay_buffer_size
-        self._params["gamma"] = gamma
 
-        self._epsilon_schedule = epsilon_schedule
-        if epsilon_schedule is None:
-            self._epsilon_schedule = ConstantSchedule(epsilon)
+        self._params = {}
+        for k, v in params.items():
+            self._params[k] = v
 
-        self._lr_schedule = lr_schedule
-        if lr_schedule is None:
-            self._lr_schedule = ConstantSchedule(lr)
+        self._lr_schedule = None
+        self.model_params = None
 
         self._model = PolicyNetwork(self._params["obs_dim"], self._params["act_dim"])
-        self._optimizer = tf.keras.optimizers.Adam(learning_rate=self._lr_schedule.get_value())
+        self._optimizer = tf.keras.optimizers.Adam(learning_rate=self._params["lr"])
+        self._replay_buffer = CircularReplayBuffer(size=self._params["max_replay_buffer_size"])
+
+        self._replay_buffer._dtype.pop("legal_moves_as_int")
+        self._replay_buffer._dtype.pop("next_legal_moves_as_int")
+
+        self._replay_buffer._data = {}
+        for data_key in self._replay_buffer._dtype:
+            self._replay_buffer._data[data_key] = [None] * int(self._replay_buffer._size)
+
+
+    def update_agent(self):
 
-        self._replay_buffer = CircularReplayBuffer(
-            seed=self._params["seed"], size=self._params["max_replay_buffer_size"]
-        )
+        if self._lr_schedule is None:
+            self._lr_schedule = ConstantSchedule(self._params["lr"])
+        else:
+            self._optimizer._lr = self._lr_schedule.get_value()
```

Comment:
It seems weird to be updating a "private" argument of the optimizer like that. did you see it as a good practice somewhere ?

the keras optimizers can be provided with a variable learning rate (cf. https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer#callable_learning_rate) you can either provide `self._lr_schedule.get_value` or just use one of their schedule class and not rely on hive's for that (I think the second option is better).